colcon_core/package_discovery/path.py

```diff
@@ -30,13 +32,14 @@ def add_arguments(
             '--paths',
             nargs='*' if not single_path else '?',
             metavar='PATH',
-            default='.' if with_default else None,
+            default=wrap_default_value(['.']) if with_default else None,
```

Comment:
Shouldn't we continue to pass a string, rather than a list?
```suggestion
            default=wrap_default_value('.') if with_default else None,
```

Reply:
The argument uses `nargs='*'` which makes the value a list. Setting the default to a string was a bug before. Luckily iterating a string with a single character has the same effect as iterating a list of strings in this case. For a different value that would be broken (see e.g. colcon/colcon-metadata#18).